tmt/steps/discover/fmf.py

```diff
@@ -147,6 +152,43 @@ def go(self):
         testdir = os.path.join(self.workdir, 'tests')
         dist_git_source = self.get('dist-git-source', False)
 
+        # Raise an exception if --fmf-id uses w/o url and git root
+        # doesn't exist for discovered plan
+        if self.opt('fmf_id'):
+            def print_error(plan_name=None):
+                try:
+                    subprocess.run(
+                        'git rev-parse --show-toplevel'.split(),
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.DEVNULL,
+                        check=True)
+                except subprocess.CalledProcessError:
+                    raise tmt.utils.GeneralError(
+                        f"`tmt run discover --fmf-id` without `url` option in "
+                        f"plan `{plan_name}` can be used only within"
+                        f" git repo.")
+
+            try:
+                fmf_tree = fmf.Tree(os.getcwd())
+            except fmf.utils.RootError:
```

Comment:
One is unable to create cmdline only plan, e.g. `tmt run -r  discover -h fmf --fmf-id --url  URL` in directory without fmf root.
IMO not existing fmf_tree is valid use case, so later `enumerate(fmf_tree.climb()` can be part of try/except and exception will be ignored.
If there is no url on CLI it will be covered by `not url` check.

Reply:
This use case might to be that silly - e.g. ad-hoc scheduling from test repo without cloning it locally first.

Reply:
Thanks Lukas, very good point. This is a valid case but I don't understand how to fix that. Could you please provide a patch to ignore exception?

Another point. I'm thinking that there is bug if we call `tmt run -r discover -h fmf` in directory without fmf root: https://cpaste.org/?0b98e395e9727532#3RGXsT1eu8fc8b3uzXZWYbk8KxGPDHNTEpN7VPQd97XY

Reply:
In all cases I'd raise `tmt.utils.DiscoverError` as discover plugin can't continue and that way tmt finishes with proper message and exit code - https://cpaste.org/?d2bb05249f808666#GKi6Ao1FMibsGa5Jjy4t4WN95bTmjNGgrZVuap3aYaPS